frontend/app/assets/javascripts/spree/frontend/checkout/coupon-code.js.coffee

```diff
@@ -0,0 +1,39 @@
+Spree.onCouponCodeApply = (e) ->
+  coupon_code_field = $('#order_coupon_code')
+  coupon_code = $.trim(coupon_code_field.val())
+  return if coupon_code == ''
+
+  if $('#coupon_status').length == 0
+    coupon_status = $("<div id='coupon_status'></div>")
+    coupon_code_field.parent().append(coupon_status)
+  else
+    coupon_status = $("#coupon_status")
+
+  successClass = coupon_status.data('success-class') || 'success'
+  errorClass = coupon_status.data('error-class') || 'error'
```

Comment:
I missed this because it was in an outdated diff. I'd lean towards agreeing with @jhawthorn here. Removing this would clean up both the JS and the markup and make the conversation below moot.